src/main/java/de/tum/in/www1/artemis/web/rest/LectureResource.java

```diff
@@ -152,28 +148,22 @@ public LectureResource(LectureRepository lectureRepository, LectureService lectu
     }
 
     /**
-     * GET /lectures/:id : get the "id" lecture.
+     * GET lectures/:lectureId : get the "id" lecture.
      *
-     * @param id the id of the lecture to retrieve
+     * @param lectureId the id of the lecture to retrieve
      * @return the ResponseEntity with status 200 (OK) and with body the lecture, or with status 404 (Not Found)
      */
-    @GetMapping("/lectures/{id}")
+    @GetMapping("lectures/{lectureId}")// TODO: should be /courses/{courseId}/lectures/{lectureId}
     @PreAuthorize("hasRole('USER')")
-    public ResponseEntity<Lecture> getLecture(@PathVariable Long id) {
-        log.debug("REST request to get Lecture : {}", id);
-        Optional<Lecture> lectureOptional = lectureRepository.findByIdWithPostsAndLectureUnitsAndLearningGoals(id);
-        if (lectureOptional.isEmpty()) {
-            return ResponseEntity.notFound().build();
-        }
-        Lecture lecture = lectureOptional.get();
+    public ResponseEntity<Lecture> getLecture(@PathVariable Long lectureId) {
+        log.debug("REST request to get Lecture : {}", lectureId);
+        Lecture lecture = lectureRepository.findByIdWithPostsAndLectureUnitsAndLearningGoalsElseThrow(lectureId);
         Course course = lecture.getCourse();
         if (course == null) {
```

Comment:
can't we use `courseRepository.findByIdElseThrow(courseId)` in here? you already used it in `updateLecture`

Reply:
I assume there was a reason why we also fetched the Posts, LectureUnits and Learning Goals from the repository here, I want to avoid this missing when it's actually needed so I just used the equivalent call but with ...ElseThrow as to change up as little as possible.